src/pocketmine/Player.php

```diff
@@ -1531,37 +1531,40 @@ public function onUpdate($currentTick){
 				}
 			}
 
-			if($this->isSurvival()){
-				if($this->speed->x > 0 or $this->speed->z > 0){
-					$this->incrementHungerTicks($this->isSprinting() ? 500 : 200);
-				}
-			}
-			if($this->getFood() <= 0){
-				$this->starveTicks++;
-				if($this->starveTicks > 20){
-					$this->starveTicks = 0;
-					if(!(
-							$this->server->getDifficulty() === 0 or
-							$this->server->getDifficulty() === 1 and $this->getHealth() <= 10 or
-							$this->server->getDifficulty() === 2 and $this->getHealth() <= 1
-					)){
-						$this->attack(1, new EntityDamageEvent($this, EntityDamageEvent::CAUSE_STARVATION, 1));
+			if($this->isSurvival() || $this->isAdventure()){
```

Comment:
isSurvival returns whether the gamemode is an odd number.
isAdventure returns whether the gamemode is 2 or 3.
The code was correct. This change is wrong.